src/battle/contest_link_80C857C.c

```diff
@@ -1,9 +1,732 @@
+            break;
+        case 10:
+            if (HaveAllPlayersReceivedBlock())
+            {
+                memcpy(gContestFinalStandings, gBlockRecvBuffer[gUnknown_0203869B], sizeof gContestFinalStandings);
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        default:
+            gTasks[taskId].data[0] = 0;
+            SwitchTaskToFollowupFunc(taskId);
+            break;
+    }
+}
+
+void sub_80C8C80(u8 taskId)
+{
+    switch (gTasks[taskId].data[0])
+    {
+        case 0:
+            if (IsLinkTaskFinished())
+            {
+                SendBlockToAllOpponents(sContestantStatus, 4 * sizeof(struct ContestantStatus));
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        case 1:
+            if (HaveAllPlayersReceivedBlock())
+            {
+                memcpy(sContestantStatus, gBlockRecvBuffer[gUnknown_0203869B], 4 * sizeof(struct ContestantStatus));
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        case 2:
+        case 5:
+        case 8:
+        case 11:
+            if (gTasks[taskId].data[1]++ > 10)
+            {
+                gTasks[taskId].data[1] = 0;
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        case 3:
+            if (IsLinkTaskFinished())
+            {
+                SendBlockToAllOpponents(&shared192D0, sizeof shared192D0);
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        case 4:
+            if (HaveAllPlayersReceivedBlock())
+            {
+                memcpy(&shared192D0, gBlockRecvBuffer[gUnknown_0203869B], sizeof shared192D0);
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        case 6:
+            if (IsLinkTaskFinished())
+            {
+                SendBlockToAllOpponents(&shared19328, sizeof shared19328);
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        case 7:
+            if (HaveAllPlayersReceivedBlock())
+            {
+                memcpy(&shared19328, gBlockRecvBuffer[gUnknown_0203869B], sizeof shared19328);
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        case 9:
+            if (IsLinkTaskFinished())
+            {
+                SendBlockToAllOpponents(gUnknown_02038696, sizeof gUnknown_02038696);
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        case 10:
+            if (HaveAllPlayersReceivedBlock())
+            {
+                memcpy(gUnknown_02038696, gBlockRecvBuffer[gUnknown_0203869B], sizeof gUnknown_02038696);
+                gTasks[taskId].data[0]++;
+            }
+            break;
+        default:
+            gTasks[taskId].data[0] = 0;
+            SwitchTaskToFollowupFunc(taskId);
+            break;
+    }
+}
+
+void sub_80C8E1C(u8 taskId)
+{
+    int i;
+
+    switch (gTasks[taskId].data[0])
+    {
+#if ENGLISH
```

Comment:
I wonder if it's better to surround the whole functions with the `#if` blocks.  Having them inline everywhere makes it illegible.

Reply:
at the very least this:
```c
#if GERMAN
                    if (gTasks[taskId].data[0] == 0)
                    {
                        gTasks[taskId].data[0] = 3;
                    }
                    else
                    {
                        de_sub_80C9274(FALSE);
#endif
                    sub_8007E9C(2);
#if ENGLISH
                    gTasks[taskId].data[0]++;
#elif GERMAN
                    gTasks[taskId].data[0] = 1;
                    }
#endif
```

```c
#if GERMAN
                    if (gTasks[taskId].data[0] == 0)
                    {
                        gTasks[taskId].data[0] = 3;
                    }
                    else
                    {
                        de_sub_80C9274(FALSE);
                        sub_8007E9C(2);
                        gTasks[taskId].data[0] = 1;
                    }
#else
                    sub_8007E9C(2);
                    gTasks[taskId].data[0]++;

#endif
```

Reply:
Agreed. Blocks shouldn't cross `#if`/`#endif` boundaries.